package/MDAnalysis/core/topologyattrs.py

```diff
@@ -2032,6 +2031,238 @@ def total_charge(group, compound='group'):
+            Choose whether the dipole moment is calculated at the center of 
+            "mass" or the center of "charge"
+
+        Returns
+        -------
+        numpy.ndarray
+            Dipole moment of (compounds of) the group in e*angstroms.
+            If `compound` was set to ``'group'``, the output will be a single
+            value. Otherwise, the output will be a 1d array of shape ``(n,)``
+            where ``n`` is the number of compounds.
+
+        .. versionadded:: 2.4.0
+        """
+
+        compound = compound.lower()
+
+        atomgroup = group.atoms
+        charges = atomgroup.charges
+
+        if center == "mass":
+            masses = atomgroup.masses
+            ref = atomgroup.center_of_mass(
+                wrap=wrap, 
+                unwrap=unwrap, 
+                compound=compound
+            )
+        elif center == "charge":
+            ref = atomgroup.center_of_charge(
+                wrap=wrap,
+                unwrap=unwrap,
+                compound=compound
+            )
+        else:
+            choices = ["mass", "charge"]
+            raise ValueError(
+                "The dipole center, {}, is not supported. Choose one of: {}".format(
+                 center,choices))
+
+        if compound == 'group':
+            if wrap:
+                recenteredpos = (atomgroup.pack_into_box(inplace=False) - ref)
+            elif unwrap:
+                recenteredpos = (atomgroup.unwrap(inplace=False) - ref)
+            else:
+                recenteredpos = (atomgroup.positions - ref)
+            dipole_moment = np.sum(recenteredpos*charges[:,np.newaxis], axis=0)
+        else:
+            (atom_masks, 
+             compound_masks, 
+             n_compounds) = atomgroup._split_by_compound_indices(compound)
+
+            if unwrap:
+                coords = atomgroup.unwrap(
+                    compound=compound,
+                    reference=None,
+                    inplace=False
+                )
+            else:
+                coords = atomgroup.positions
+            chgs = atomgroup.charges
+
+            dipole_moment = np.empty((n_compounds, 3), dtype=np.float64)
+            for compound_mask, atom_mask in zip(compound_masks, atom_masks):
+                dipole_moment[compound_mask] = np.sum(
+                    (coords[atom_mask]-ref[compound_mask][:,None,:])
+                    *chgs[atom_mask][:,:,None], axis=1)
+
+        return dipole_moment
+
+    transplants[GroupBase].append(
+        ('dipole_moment', dipole_moment))
+
+    @warn_if_not_unique
+    @_pbc_to_wrap
+    @check_wrap_and_unwrap
+    def quadrupole_moment(group, wrap=False, unwrap=False, 
+        compound='group', center="mass"):
+        r"""Quadrupole vector of the group according to [Gray1984]_
+         
+        .. math::
+            Q = \sqrt{\frac{2}{3}\boldsymbol{\hat{Q}}:\boldsymbol{\hat{Q}}}
+
+        where :math:`\boldsymbol{\hat{Q}}` is the traceless quadrupole 
```

Comment:
define $\hat{\mathsf{Q}}$.

I *think* throughout we write tensors in `\mathsf` so

$$
Q = \sqrt{\frac{2}{3}{\hat{\mathsf{Q}}}:{\hat{\mathsf{Q}}}}
$$
```latex
Q = \sqrt{\frac{2}{3}{\hat{\mathsf{Q}}}:{\hat{\mathsf{Q}}}}
```

Also write out the cartesian formulation — it avoids having to figure out the meaning of `Q:Q` etc.